src/rgw/rgw_auth.cc

```diff
@@ -535,7 +535,33 @@ void rgw::auth::LocalApplier::load_acct_info(const DoutPrefixProvider* dpp, RGWU
   user_info = this->user_info;
 }
 
-void rgw::auth::LocalApplier::modify_request_state(const DoutPrefixProvider *dpp, req_state* s) const
+void rgw::auth::RoleApplier::to_str(std::ostream& out) const {
+  out << "rgw::auth::LocalApplier(role name =" << role_name;
+  for (auto policy : role_policies) {
+    out << ", role policy =" << policy;
+  }
+  out << ")";
+}
+
+bool rgw::auth::RoleApplier::is_identity(const idset_t& ids) const {
+  for (auto& p : ids) {
+    string name = p.get_tenant() + "$" + p.get_id();
```

Comment:
if tenant is empty, should this also match 'id' by itself? or is it documented that the principal always needs the dollar sign?

Reply:
No, actually a principal can be without a tenant (and without a dollar sign), will fix this.